src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java

```diff
@@ -0,0 +1,61 @@
+package edu.kis.powp.jobs2d.drivers.adapter;
+
+import edu.kis.legacy.drawer.panel.DrawPanelController;
+import edu.kis.legacy.drawer.shape.ILine;
+import edu.kis.legacy.drawer.shape.LineFactory;
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.factiories.SpecialLineFactory;
+
+public class LineDrawerAdapter implements Job2dDriver {
+
+    private DrawPanelController controller;
+    private int x = 0;
+    private int y = 0;
+    private ILine lineToDraw = null;
+
+    public LineDrawerAdapter(DrawPanelController controller, ILine lineToDraw) {
+        this.controller = controller;
+        this.lineToDraw = lineToDraw;
+    }
+
+    public LineDrawerAdapter(DrawPanelController controller, LineType type) {
```

Comment:
I would recommend simpler solutions (KISS principle ;) ). The first constructor is sufficient. This one is redundant and introduces new dependencies.

Reply:
Fixed in latest commit to master branch.